src/types/coord.rs

```diff
@@ -29,22 +29,64 @@ where
     pub m: Option<T>,
 }
 
-impl<T> fmt::Display for Coord<T>
-where
-    T: WktNum + fmt::Display,
-{
-    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        write!(f, "{} {}", self.x, self.y)?;
-        if let Some(z) = self.z {
-            write!(f, " {}", z)?;
+macro_rules! impl_display_for_float {
+    ($t: ident) => {
+        impl fmt::Display for Coord<$t> {
+            fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+                let mut buffer = ryu::Buffer::new();
+                let x = buffer.format(self.x);
+
+                let mut buffer = ryu::Buffer::new();
+                let y = buffer.format(self.y);
+
+                write!(f, "{} {}", x, y)?;
```

Comment:
Nit: I suspect a sequence of `f.write_str` calls would be a little faster here.

Reply:
Possibly, it might muddy the waters on perf improvements though. I'd prefer to keep everything except ryu the same in this PR.